### PR TITLE
Add surrounding whitespace work title cleanup bot

### PR DESCRIPTION
**Pull request by jimman2003**
## Description:
In this Pull Request we have made the following changes:
* Added a remix of making_a_bot.py but for works

### CODE REVIEW

File: whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py [hunk 1]
```diff
@@ -17,20 +16,20 @@ def run(self) -> None:  # overwrite the AbstractBotJob run method
...
                 # extract info from the dump file and check it
                 row, json_data = self.process_row(row)
-                if json_data['type']['key'] != '/type/work':
+                if json_data["type"]["key"] != "/type/work":
                     continue  # this can be done faster with a grep filter, but for this example we'll do it here
-                if not self.needs_trim(json_data['title']):
+                if not self.needs_trim(json_data["title"]):
                     continue
 
                 # the database may have changed since the dump was created, so call the OpenLibrary API and check again
```
Comment (cclauss) [inline whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py:29]: 88 characters per line max please

File: whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py [hunk 2]
```diff
@@ -39,12 +38,12 @@ def run(self) -> None:  # overwrite the AbstractBotJob run method
...
                 work.title = work.title.strip()
                 self.logger.info(
-                    '|'.join([olid, old_title, work.title])
+                    "|".join([olid, old_title, work.title])
                 )  # don't forget to log modifications!
                 self.save(lambda: work.save(comment=comment))
 
 
-if '__name__' == '__name__':
+if "__name__" == "__main__":
```
Comment (cclauss) [inline whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py:46]:
```suggestion
if __name__ == "__main__”:
```
Dunder name is a builtin variable in Python.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py (hunk 1 above)
Comment (jimman2003) [reply whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py:29]: " is one character

File: whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    continue  # skip deleted editions
+                if not self.needs_trim(work.title):
+                    continue
+
+                # this edition needs editing, so fix it
+                old_title = copy.deepcopy(work.title)
+                work.title = work.title.strip()
+                self.logger.info(
+                    "|".join([olid, old_title, work.title])
+                )  # don't forget to log modifications!
```
Comment (cclauss) [inline whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py:43]:
```suggestion
                self.logger.info("|".join((olid, old_title, work.title)))  # log the modifications
```

File: whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+if __name__ == "__main__":
+    job = TrimTitleJob()
+
+    try:
+        job.run()
+    except Exception as e:
+        job.logger.exception("")
+        raise e
```
Comment (cclauss) [inline whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py:54]:
```suggestion
    except Exception:
        job.logger.exception("TrimTitleJob failed")
        raise
```
Or  `job.logger.exception("whitespace_work_title_cleanup failed")`

**Review by cclauss — CHANGES_REQUESTED** [review]

File: whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import copy
```
Comment (cclauss) [inline whitespace_work_title_cleanup_bot/whitespace_work_title_cleanup.py:1]:
```suggestion
#!/usr/bin/env python3

import copy
```

**Review by cclauss — APPROVED** [review]
Thanks much.

**Pull request merged**